File: Weather/Weather/Data Models/WeatherForecast.swift
//
//  ForcastWeather.swift
//  Weather
//
//  Created by Pasan Diksura on 2024-11-23.
//

import Foundation

struct WeatherForecastDTO: Decodable, Hashable {
    let location: LocationDTO
    let current: CurrentDTO
    let forecast: ForecastWeather
    let alerts: AlertsDTO
}

struct ForecastWeather: Decodable, Hashable {
    let forecastday: [ForecastDayDTO]
}

File: WeatherServer/WeatherServer.API/Models/WeatherForecast.cs
namespace WeatherServer.API.Models
{
    public record WeatherForecast
        (
            Location location,
            Current current,
            ForecastWeather forecast,
            Alerts alerts
        );

    public record ForecastWeather
        (
            List<ForecastDay> forecastday
        );
}
